src/api/mod.rs

```diff
@@ -174,6 +184,112 @@ impl Api {
             }
         });
     }
+
+    fn start_eth_background_task(
```

Comment:
we should have a `SubstrateApi` and a `EthApi` and maybe a BaseAPI for shared logics
mixing logics is just making everything confusing